plugins/rest/rest_auth.go

```diff
@@ -155,15 +155,17 @@ type tokenEndpointResponse struct {
 // oauth2ClientCredentialsAuthPlugin represents authentication via a bearer token in the HTTP Authorization header
 // obtained through the OAuth2 client credentials flow
 type oauth2ClientCredentialsAuthPlugin struct {
-	GrantType    string                 `json:"grant_type"`
-	TokenURL     string                 `json:"token_url"`
-	ClientID     string                 `json:"client_id"`
-	ClientSecret string                 `json:"client_secret"`
-	SigningKeyID string                 `json:"signing_key"`
-	Thumbprint   string                 `json:"thumbprint"`
-	Claims       map[string]interface{} `json:"additional_claims"`
-	IncludeJti   bool                   `json:"include_jti_claim"`
-	Scopes       []string               `json:"scopes,omitempty"`
+	GrantType            string                 `json:"grant_type"`
+	TokenURL             string                 `json:"token_url"`
+	ClientID             string                 `json:"client_id"`
+	ClientSecret         string                 `json:"client_secret"`
+	SigningKeyID         string                 `json:"signing_key"`
+	Thumbprint           string                 `json:"thumbprint"`
+	Claims               map[string]interface{} `json:"additional_claims"`
+	IncludeJti           bool                   `json:"include_jti_claim"`
+	Scopes               []string               `json:"scopes,omitempty"`
+	AdditionalHeaders    map[string]interface{} `json:"additional_headers,omitempty"`
+	AdditionalParameters map[string]interface{} `json:"additional_parameters,omitempty"`
```

Comment:
nit: I think we can use `map[string]string` here, as both headers and POST params have no other type, and it's not clear to me what happens if someone provides a map, list, or whatever else here, and we could skip the sprintf conversion below.